webapp/src/modules/asset/utils.spec.ts

```diff
@@ -56,3 +66,205 @@ describe("when getting the asset's url", () => {
+      })
+      it('should return true', () => {
+        expect(isOwnedBy(asset, wallet)).toBe(true)
+      })
+    })
+    describe('and the logged user is not the asset owner', () => {
+      beforeEach(() => {
+        wallet = {
+          address: 'notTheAssetOwner'
+        } as Wallet
+      })
+      it('should return false', () => {
+        expect(isOwnedBy(asset, wallet)).toBe(false)
+      })
+    })
+  })
+
+  describe('and the asset has a rental', () => {
+    describe('and the rental is in status OPEN', () => {
+      beforeEach(() => {
+        asset = {
+          openRentalId: null,
+          owner: 'anOwner'
+        } as Asset
+        rental = {
+          status: RentalStatus.OPEN
+        } as RentalListing
+      })
+
+      describe('and the logged user is the asset owner', () => {
+        beforeEach(() => {
+          asset = {
+            openRentalId: null,
+            owner: 'anOwner'
+          } as Asset
+          wallet = {
+            address: (asset as NFT).owner
+          } as Wallet
+        })
+        it('should return true', () => {
+          expect(isOwnedBy(asset, wallet, rental)).toBe(true)
+        })
+      })
+
+      describe('and the logged user is not the asset owner', () => {
+        beforeEach(() => {
+          asset = {
+            openRentalId: null,
+            owner: 'anOwner',
+            chainId: 1
+          } as Asset
+          wallet = {
+            address: 'notTheAssetOwner'
+          } as Wallet
+        })
+        it('should return false', () => {
+          expect(isOwnedBy(asset, wallet, rental)).toBe(false)
+        })
+      })
+    })
+
+    describe('and the rental is in status EXECUTED', () => {
+      describe('and the logged user is the rental lessor the asset is in the rentals contract', () => {
+        const rentalsContract: ContractData = getContract(
+          ContractName.Rentals,
+          1
+        )
+        beforeEach(() => {
+          asset = {
+            openRentalId: null,
+            owner: rentalsContract.address,
+            chainId: 1
+          } as Asset
+          wallet = {
+            address: 'anAddress'
+          } as Wallet
+          rental = {
+            status: RentalStatus.EXECUTED,
+            lessor: wallet.address
+          } as RentalListing
+        })
+        it('should return true', () => {
+          expect(isOwnedBy(asset, wallet, rental)).toBe(true)
+        })
+      })
+
+      describe('and the logged user is not the rental lessor', () => {
+        beforeEach(() => {
+          asset = {
+            openRentalId: null,
+            owner: 'anOwner',
+            chainId: 1
+          } as Asset
+          wallet = {
+            address: 'notTheRentalLessor'
+          } as Wallet
+        })
+        it('should return false', () => {
+          expect(isOwnedBy(asset, wallet, rental)).toBe(false)
+        })
+      })
+    })
+
+    describe('and the rental is in status CANCELLED', () => {
```

Comment:
As this describe, describes the context of execution, you might want to use a `beforeEach` to specify that all the tests executed under this context will have the rental as CANCELLED.
```suggestion
    describe('and the rental is in status CANCELLED', () => {
      beforeEach(() => {
        rental.status = RentalStatus.CANCELLED
      })
```